src/zerocoin_v3.cpp

```diff
@@ -754,10 +830,31 @@ bool CSigmaState::IsUsedCoinSerial(const Scalar &coinSerial) {
     return usedCoinSerials.count(coinSerial) != 0;
 }
 
+bool CSigmaState::IsUsedCoinSerialHash(Scalar &coinSerial, const uint256 &coinSerialHash) {
+    for ( auto it = usedCoinSerials.begin(); it != usedCoinSerials.end(); ++it ){
+        if(GetSerialHash(*it)==coinSerialHash){
+            coinSerial = *it;
+            return true;
+        }
+    }
+    return false;
+}
+
 bool CSigmaState::HasCoin(const sigma::PublicCoin& pubCoin) {
     return mintedPubCoins.find(pubCoin) != mintedPubCoins.end();
 }
 
+bool CSigmaState::HasCoinHash(GroupElement &pubCoinValue, const uint256 &pubCoinValueHash) {
+    for ( auto it = mintedPubCoins.begin(); it != mintedPubCoins.end(); ++it ){
+    	sigma::PublicCoin pubCoin = (*it).first;
```

Comment:
Better to use const reference here. No need to copy.

Reply:
Do you mean `const` for `pubCoinValue`?

Reply:
You can just write const sigma::PublicCoin& pubCoin here..

Reply:
Sure